sway-parse/src/parse.rs

```diff
@@ -146,8 +146,7 @@ impl Parse for Ident {
                 }
 
                 if !ident.is_raw_ident()
-                    && (RESERVED_KEYWORDS.contains(ident_str)
-                        || (parser.experimental.error_type && ident_str == PanicToken::AS_STR))
+                    && (RESERVED_KEYWORDS.contains(ident_str) || ident_str == PanicToken::AS_STR)
```

Comment:
`RESERVED_KEYWORDS` now contain `panic` keyword, so the other part of the check is redundant.
```suggestion
                    && RESERVED_KEYWORDS.contains(ident_str)
```